modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/AbstractParameterResolverExecutor.java

```diff
@@ -87,9 +94,13 @@ protected ParameterValueResolver parameterValueResolver(ParameterizedElementDecl
         final ParameterModel parameterModel = parameterGroupModel.getParameter(parameterName)
             .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find parameter with name: '%s' in parameter group: '%s'",
                                                                             parameterName, parameterGroupName)));
-        parametersMap.put(parameterName,
-                          extractValue(parameterElement.getValue(),
-                                       artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration)));
+        Object value = extractValue(parameterElement.getValue(),
+                                    artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration));
+        if (!parameterModel.getExpressionSupport().equals(NOT_SUPPORTED) && isExpression(value)) {
```

Comment:
Even if the parameterModel supportExpressions. Shouldn't we fail? I mean, we won't be able to process them anyways.

Reply:
Well, in that case the "value" is handled as String and it is up to the resolver to handle it. I added this in order to prevent doing the parsing of the String for each parameter as it has impact in performance.